src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,166 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            addFirst(value);
+        } else {
+            addLast(value);
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, size + 1);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (tail == null && head == null) {
+            addFirst(value);
```

Comment:
in case if `index == 0` and both `head` and `tail` are null these 2 `if` statements are the same

so basically you can remove the second `if` statement

Reply:
not fixed